### PR TITLE
Add dataset field level tags to UI

### PR DESCRIPTION
**Pull request by davidsharp7**
### Problem

Currently the UI has no ability to add and delete tags at the field level. The proposed solution is update the DatasetTags component to allow for field level tagging/deletion and add this to the DatasetInfo component.

Closes: #2728 

### Solution

Screenshot 2024-02-09 at 07 07 16

Screenshot 2024-02-01 at 18 18 19

Update DatasetTags to include field level tagging in the UI.

One-line summary:

Update DatasetTags to include field level tagging in the UI

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [x] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [x] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [x] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

**Review by wslulciuc — COMMENTED** [review]
LGTM 💯, but let's also get a +1 from @phixMe

**Review by wslulciuc — APPROVED** [review]

File: web/src/components/datasets/DatasetInfo.tsx
Comment (phixMe) [inline web/src/components/datasets/DatasetInfo.tsx:84]: I kind of want to get away from doing this because it doesn't move across urls. Could we possibly use use parameters for functionality like this?

File: web/src/components/datasets/DatasetInfo.tsx
Comment (davidsharp7) [reply web/src/components/datasets/DatasetInfo.tsx:157]: Cool - thanks for the feedback.  Yeah I agree it's not as slick as I would like.  I have refactored and DatasetTags now looks like this:

Screenshot 2024-02-01 at 18 20 19

Screenshot 2024-02-01 at 18 20 48

**Review by phixMe — APPROVED** [review]
Thanks @davidsharp7 This looks good now. We're going to have a merge conflict with the larger changes across the board, I may want to hold off merging this unless you need it immediately.

**Comment by davidsharp7** [comment]
> Thanks @davidsharp7 This looks good now. We're going to have a merge conflict with the larger changes across the board, I may want to hold off merging this unless you need it immediately.

Cool, cool - No giant rush from me @phixMe @wslulciuc .

**Pull request merged**
